### PR TITLE
fix(benchmarks): cap rule-discovery search ints before hang

### PR DESCRIPTION
**Pull request by ss251**
## Summary

Occupancy-FREE complete hang on `origin/main` `ec035f73ad8e1e0545155e94c3e689d4b8d5d82b`.

`_require_search_int` was type-exact with a minimum and **no maximum**. Public `run_search` / CLI `--n-random` fed that int into `random_genomes(key, n)` as shape `(n, GENOME_SIZE)` float32 (`GENOME_SIZE=33`). `tune_champion_baseline` does the same, then `range(generations)`.

```text
# red on origin/main ec035f73
_require_search_int("n_random", 10**12, minimum=0) == 1000000000000
# 10**12 * 33 * 4 = 132_000_000_000_000 bytes
run_search(n_random=10**12, ...) reached seed_genomes / random_genomes
```

That is the hang/OOM class. Public last-fit is CLI `--n-random 3072`. This PR rejects `1_000_001` and `10**12` rather than blessing leftover INT32 persist.

Not `#1874` (`rule_discovery_summary.py` only). Not clocks / forager / IA / FTL. Not `#2005` / `#1989` / JSON-nest / jax.tree / SIGReg nest.

## Expected behavior

Search ints in `[minimum, 1_000_000]` still pass. `10**12` and `1_000_001` raise `ValueError` matching `exceeds the search integer limit` before `random_genomes` / stream materialization.

## Scope

- `alberta_framework/benchmarks/rule_discovery.py`
- `tests/test_rule_discovery_search_int_hang.py`

## Verification

```text
# red on origin/main ec035f73
_require_search_int("n_random", 10**12, minimum=0) == 1000000000000

# green at this head
.venv/bin/python -m pytest tests/test_rule_discovery_search_int_hang.py tests/test_rule_discovery_hostile_validation.py tests/test_rule_discovery.py -q -o addopts=""
# 87 passed
.venv/bin/python -m ruff check alberta_framework/benchmarks/rule_discovery.py tests/test_rule_discovery_search_int_hang.py
.venv/bin/python -m mypy alberta_framework/benchmarks/rule_discovery.py tests/test_rule_discovery_search_int_hang.py
```

- [x] logs: origin `ec035f73` `_require_search_int("n_random", 10**12)` returned `1000000000000` and `run_search` reached genome generation; this head rejects `10**12` / `1_000_001` before `random_genomes`; 6 hang tests + 81 existing rule-discovery tests passed (87)

- [x] domain-artifact: N/A - search-int hang preflight, no IPMNIST shard

- [x] llm-trajectory: N/A - local fail-before-allocation proof, no model-trajectory artifact

- AI assistance: yes

- Model(s) used: xai/grok-4.6

- Agent tooling: Grok CLI via cmux

- Skill path: elizaOS/slopdotcash@6000f7847387c3f046d2c4df071a7c0bb8ddc8d6:skills/contribute-to-asi

- Provenance status: self-reported

AI provider/model: xAI / grok-4.6
Client / agent tooling: Grok CLI via cmux
Contribution skill revision: elizaOS/slopdotcash@6000f7847387c3f046d2c4df071a7c0bb8ddc8d6:skills/contribute-to-asi
Attribution status: self-reported
— [grok-cli-ss251]

### CODE REVIEW

**Comment by lalalune** [comment]
Closing after comparison with the reviewed rule-discovery fix in #2049. This patch adds one arbitrary global integer cap (1,000,000), but that still permits catastrophic candidate × stream × seed work and rejects unrelated dimensions without regard to allocation shape. #2049 instead preflights direct allocations, aggregate search/baseline/holdout logical work, retained stream bytes, exact config identities, and evolution edge cases before JAX allocation/execution. The issue is real; the fixed integer ceiling is the surface-level version and is superseded by the resource-based fix.
